File: minseon/23_01_12_02.py
# 최대공약수와 최소공배수
import math

#
# def solution(n, m):
#     answer = []
#     max = []
#     min = 1
#     max2 = 1
#     print("-----------처음 시작---------------")
#     print(f'max: {max}')
#     print(f'n : {n} , m: {m}')
#     for a in range(1,n+1):
#         while a>1 and n % a == 0 and m % a == 0:
#             max.append(a)
#             n = n//a
#             m = m//a
#             print(f'-----------와일문{a}---------------')
#             print(f'max: {max}')
#             print(f'n : {n} , m: {m}')
#             if a == n:
#                 break
#     print("-----------포문 끝---------------")
#     print(f'max: {max}')
#     print(f'n : {n} , m: {m}')
#
#     for x in max:
#         max2 = max2*x
#         print(f'max2 : {max2}')
#     min = max2*m*n
#     print(f'min : {min}')
#     answer.append(max2)
#     answer.append(min)
#     return answer
#
# print(solution(6, 12))


#유클리드 호제법을 사용한 최대공약수, 최소공배수 구하기
#
def solution(a, b):
    c, d = max(a, b), min(a, b)
    t = 1
    while t > 0:
        t = c % d
        c = d
        d = t
    # 최소 공배수 구하는 공식
    answer = [c, int(a*b/c)]

    return answer

print(solution(6, 12))
#

File: sein/23_01_12_02.java
/*
문제 설명
두 수를 입력받아 두 수의 최대공약수와 최소공배수를 반환하는 함수, solution을 완성해 보세요. 배열의 맨 앞에 최대공약수, 그다음 최소공배수를 넣어 반환하면 됩니다. 예를 들어 두 수 3, 12의 최대공약수는 3, 최소공배수는 12이므로 solution(3, 12)는 [3, 12]를 반환해야 합니다.

제한 사항
두 수는 1이상 1000000이하의 자연수입니다.
*/

class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        
        for(int i=1; i<=m; i++){
            if(n%i==0 && m%i==0){
                answer[0] = i;          //최대공약수 
            }
        }
        answer[1] = (n*m)/answer[0];  //최소공배수
        return answer;
    }
}
